mods/default/nodes.lua

```diff
@@ -737,6 +737,35 @@ minetest.register_node("default:apple", {
 	after_place_node = function(pos, placer, itemstack)
 		minetest.set_node(pos, {name = "default:apple", param2 = 1})
 	end,
+
+	after_dig_node = function(pos, oldnode, oldmetadata, digger)
+		if oldnode.param2 == 0 then
+			minetest.set_node(pos, {name = "default:apple_mark"})
+			minetest.get_node_timer(pos):start(math.random(2400, 4800))
```

Comment:
Maybe it should be similar to sapling growth time? Longer seems weird in a way, see https://github.com/minetest/minetest_game/blob/ace7ec953ea492c7c641dfeb2ce22985956fb7db/mods/default/nodes.lua#L673

Reply:
No objections to adjusting, but I thought you wanted it to be in-game days. Just let me know what you prefer.

Reply:
Yes sorry, i later realised that sapling growth time may be shorter than the timespan i seemed to suggest. It would be odd for apples to regrow slower than a tree grows, what do you think?
So since it seems best to have a fairly long apple grow time it seems best to set it to the same as sapling grow time.